osu.Game/Screens/Select/Carousel/DrawableCarouselBeatmap.cs

```diff
@@ -134,6 +157,14 @@ private void load(SongSelect songSelect, BeatmapManager manager, BeatmapSetOverl
             };
         }
 
+        private ScoreInfo getTopScore(ScoreManager scoreManager, int beatmapID, int rulesetID)
+        {
+            return scoreManager
+                .QueryScores(s => !s.DeletePending && s.Beatmap.ID == beatmapID && s.Ruleset.ID == rulesetID)
```

Comment:
As per the other comment, this should display scores from the user's selected ruleset rather than the beatmap's ruleset.